impl/src/main/java/org/jboss/weld/Container.java

```diff
@@ -40,6 +41,8 @@
 
     private static Singleton<Container> instance;
 
+    private static Environment environment;
```

Comment:
This field can be accessed concurrently so I would use `AtomicReference` and `AtomicReference#compareAndSet()` in the `setEnvironment()` method below.

Reply:
good point